core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -483,6 +510,20 @@ void injectDependencies(BaseCommand baseCommand) {
         } while (!clazz.equals(BaseCommand.class));
     }
 
+    protected <T> void reinjectDependencies(Class<? extends T> type, Class<?> owner, Object instance, T newValue) {
+        Set<MethodHandle> handles = dependencyFields.get(type, owner);
+        if (handles.isEmpty()) {
+            return;
+        }
+        for (MethodHandle handle : handles) {
+            try {
```

Comment:
per above comment, we must only match on keys. It's prob easiest to just use Fields instead of MethodHandles